src/Network/Network/ApplicationGateway/GetAzureApplicationGatewayAvailableServerVariableAndHeader.cs

```diff
@@ -0,0 +1,98 @@
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using AutoMapper;
+using Microsoft.Azure.Commands.Common.Authentication;
+using Microsoft.Azure.Commands.Common.Authentication.Abstractions;
+using Microsoft.Azure.Commands.Network.Models;
+using Microsoft.Azure.Management.Network;
+using Microsoft.WindowsAzure.Commands.Common;
+using System.Collections.Generic;
+using System.IO;
+using System.Management.Automation;
+using System.Reflection;
+
+namespace Microsoft.Azure.Commands.Network
+{
+    [Cmdlet("Get", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "ApplicationGatewayAvailableServerVariableAndHeader"), OutputType(typeof(PSApplicationGatewayAvailableServerVariableAndRequestHeaderResult))]
+    [Alias("List-" + ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "ApplicationGatewayAvailableServerVariableAndHeader")]
+    public class GetAzureApplicationGatewayAvailableServerVariableAndHeader : ApplicationGatewayBaseCmdlet
+    {
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Application Gateway available server variables.")]
+        public SwitchParameter ServerVariables {
+            get { return isServerVariableRequested; }
+            set { isServerVariableRequested = value; }
+        }
+        private bool isServerVariableRequested;
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Application Gateway available request headers.")]
+        public SwitchParameter RequestHeaders {
+            get { return isRequestHeaderRequested; }
+            set { isRequestHeaderRequested = value; }
+        }
+        private bool isRequestHeaderRequested;
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Application Gateway available response headers.")]
+        public SwitchParameter ResponseHeaders {
+            get { return isResponseHeaderRequested; }
+            set { isResponseHeaderRequested = value; }
+        }
+        private bool isResponseHeaderRequested;
+
+        public override void ExecuteCmdlet()
+        {
+            base.ExecuteCmdlet();
+            IList<string> availableServerVariable;
+            IList<string> availableRequestHeader;
+            IList<string> availableResponseHeader;
+            Dictionary<string, IList<string>> AvailableServerVariableAndHeaderResponse = new Dictionary<string, IList<string>>();
+
+            if (!isServerVariableRequested && !isRequestHeaderRequested && !isResponseHeaderRequested)
```

Comment:
I get the idea, but this creates duplicate code. Maybe it would be better to have a variable "get all" (not sure about the name) and check like this:

```c#
bool gelAll = !this.ServerVariables.IsPresent && !.. & !..;

if(this.RequestHeaders || getAll) { //do stuff }
if(this.ResponseHeaders || getAll) { //do stuff }
// and so on
```

Reply:
Removed the duplicate code.